craft_parts/infos.py

```diff
@@ -139,6 +141,69 @@ def _set_machine(self, arch: Optional[str]):
         self._machine = machine
 
 
+class PartInfo:
+    """Part-level information containing project and part fields.
+
+    :param project_info: The project information.
+    :param part: The part we want to obtain information from.
+    """
+
+    def __init__(
+        self,
+        project_info: ProjectInfo,
+        part: Part,
+    ):
+        self._project_info = project_info
+        self._part_name = part.name
+        self._part_src_dir = part.part_src_dir
+        self._part_src_subdir = part.part_src_subdir
+        self._part_build_dir = part.part_build_dir
+        self._part_build_subdir = part.part_build_subdir
+        self._part_install_dir = part.part_install_dir
+        self._part_state_dir = part.part_state_dir
+
+    def __getattr__(self, name):
```

Comment:
please document why, this is strange :-)

Reply:
The final goal is to make `ProjectInfo` attributes available to `PartInfo`. The *Info classes use composition with attribute cascading to avoid setting a lot of attributes in the constructor (and missing something in the process). I can also implement it that way if you think inheritance is a better choice here. There will be one more class, `StepInfo`, that adds to that sequence (`ProjectInfo` -> `PartInfo` -> `StepInfo`). The info classes are not used internally and are just a way to feed relevant data to pre/post-step callbacks.

Reply:
Please do use inheritance :D It doesn't matter if there are a lot of attrs.